// @intacct/intacct-sdk/dist/Functions/Common/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NewQuery = exports.Query = exports.GetList = exports.Lookup = exports.Inspect = exports.ReadMore = exports.ReadByQuery = exports.ReadByName = exports.Read = void 0;
var Read_1 = require("./Read");
Object.defineProperty(exports, "Read", { enumerable: true, get: function () { return Read_1.default; } });
var ReadByName_1 = require("./ReadByName");
Object.defineProperty(exports, "ReadByName", { enumerable: true, get: function () { return ReadByName_1.default; } });
var ReadByQuery_1 = require("./ReadByQuery");
Object.defineProperty(exports, "ReadByQuery", { enumerable: true, get: function () { return ReadByQuery_1.default; } });
var ReadMore_1 = require("./ReadMore");
Object.defineProperty(exports, "ReadMore", { enumerable: true, get: function () { return ReadMore_1.default; } });
var Inspect_1 = require("../Common/Inspect");
Object.defineProperty(exports, "Inspect", { enumerable: true, get: function () { return Inspect_1.default; } });
var Lookup_1 = require("../Common/Lookup");
Object.defineProperty(exports, "Lookup", { enumerable: true, get: function () { return Lookup_1.default; } });
const GetList = require("./GetList/index");
exports.GetList = GetList;
const Query = require("./Query/index");
exports.Query = Query;
const NewQuery = require("./NewQuery/index");
exports.NewQuery = NewQuery;
