// @intacct/intacct-sdk/dist/Functions/Projects/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimesheetEntryUpdate = exports.TimesheetEntryCreate = exports.TimesheetDelete = exports.TimesheetDecline = exports.TimesheetUpdate = exports.TimesheetCreate = exports.TimesheetApprove = exports.TaskUpdate = exports.TaskDelete = exports.TaskCreate = exports.ProjectUpdate = exports.ProjectDelete = exports.ProjectCreate = exports.AbstractTimesheetEntry = exports.AbstractTimesheet = exports.AbstractTask = exports.AbstractProject = void 0;
var AbstractProject_1 = require("./AbstractProject");
Object.defineProperty(exports, "AbstractProject", { enumerable: true, get: function () { return AbstractProject_1.default; } });
var AbstractTask_1 = require("./AbstractTask");
Object.defineProperty(exports, "AbstractTask", { enumerable: true, get: function () { return AbstractTask_1.default; } });
var AbstractTimesheet_1 = require("./AbstractTimesheet");
Object.defineProperty(exports, "AbstractTimesheet", { enumerable: true, get: function () { return AbstractTimesheet_1.default; } });
var AbstractTimesheetEntry_1 = require("./AbstractTimesheetEntry");
Object.defineProperty(exports, "AbstractTimesheetEntry", { enumerable: true, get: function () { return AbstractTimesheetEntry_1.default; } });
var ProjectCreate_1 = require("./ProjectCreate");
Object.defineProperty(exports, "ProjectCreate", { enumerable: true, get: function () { return ProjectCreate_1.default; } });
var ProjectDelete_1 = require("./ProjectDelete");
Object.defineProperty(exports, "ProjectDelete", { enumerable: true, get: function () { return ProjectDelete_1.default; } });
var ProjectUpdate_1 = require("./ProjectUpdate");
Object.defineProperty(exports, "ProjectUpdate", { enumerable: true, get: function () { return ProjectUpdate_1.default; } });
var TaskCreate_1 = require("./TaskCreate");
Object.defineProperty(exports, "TaskCreate", { enumerable: true, get: function () { return TaskCreate_1.default; } });
var TaskDelete_1 = require("./TaskDelete");
Object.defineProperty(exports, "TaskDelete", { enumerable: true, get: function () { return TaskDelete_1.default; } });
var TaskUpdate_1 = require("./TaskUpdate");
Object.defineProperty(exports, "TaskUpdate", { enumerable: true, get: function () { return TaskUpdate_1.default; } });
var TimesheetApprove_1 = require("./TimesheetApprove");
Object.defineProperty(exports, "TimesheetApprove", { enumerable: true, get: function () { return TimesheetApprove_1.default; } });
var TimesheetCreate_1 = require("./TimesheetCreate");
Object.defineProperty(exports, "TimesheetCreate", { enumerable: true, get: function () { return TimesheetCreate_1.default; } });
var TimesheetUpdate_1 = require("./TimesheetUpdate");
Object.defineProperty(exports, "TimesheetUpdate", { enumerable: true, get: function () { return TimesheetUpdate_1.default; } });
var TimesheetDecline_1 = require("./TimesheetDecline");
Object.defineProperty(exports, "TimesheetDecline", { enumerable: true, get: function () { return TimesheetDecline_1.default; } });
var TimesheetDelete_1 = require("./TimesheetDelete");
Object.defineProperty(exports, "TimesheetDelete", { enumerable: true, get: function () { return TimesheetDelete_1.default; } });
var TimesheetEntryCreate_1 = require("./TimesheetEntryCreate");
Object.defineProperty(exports, "TimesheetEntryCreate", { enumerable: true, get: function () { return TimesheetEntryCreate_1.default; } });
var TimesheetEntryUpdate_1 = require("./TimesheetEntryUpdate");
Object.defineProperty(exports, "TimesheetEntryUpdate", { enumerable: true, get: function () { return TimesheetEntryUpdate_1.default; } });
